src/backend/api/meals.js

```diff
@@ -3,13 +3,149 @@ const router = express.Router();
 const knex = require("../database");
 
 router.get("/", async (request, response) => {
-  try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
-  } catch (error) {
-    throw error;
+    //maxPrice
+    if("maxPrice" in request.query){
+      const maxPrice = Number(request.query.maxPrice);
+      if(isNaN(maxPrice)){
+        response.send('maxPrice should be a number!')
+      }else{
+         meals = knex("meal").where("price", "<", maxPrice);
+      }
+    }
+     //availableReservations
+    if("availableReservations" in request.query){
+      const reqAvailReservation = request.query.availableReservations;
+      if(reqAvailReservation != 'true'){
+        response.send('enter true value of availableReservations!')
+      }else {
+        meals = knex.raw(`select meal.id , meal.title, (meal.max_reservations)-(reservation.number_of_guests) AS AvailableReservation from meal inner join reservation on meal.id = reservation.meal_id where ((max_reservations)-(number_of_guests)) > 0`)
+      }
+    }
+     //title
+    if("title" in request.query) {
+      const reqTitle = request.query.title;
+      if(!isNaN(reqTitle)){
+        response.send("Title should be a string! ")
+      }else{
+        meals = knex('meal').where("title", "like", `%${reqTitle}%`)
+      }
+    }
+    //dateAfter
+    if("dateAfter" in request.query){
+      const reqDate = new Date(request.query.dateAfter)
+      meals =  knex('meal').where("when_date", ">", reqDate )
```

Comment:
Is the field called `when_date` in your database? In the shared DB diagram it is just called `when`: https://dbdiagram.io/d/5f0460690425da461f045a29

Reply:
Yes I don't know how but in my database it is when_date, 
I can update the name of column  but then have to do lots of work with all queries  🤔

Reply:
It is actually a good choice to update to when_date, because 'when' is a reserved word in SQL queries and it caused some issues for me. I needed to use `when` for queries to work.